packages/sign-client/src/controllers/engine.ts

```diff
@@ -886,6 +964,13 @@ export class Engine extends IEngine {
       throw new Error(`Could not find pending auth request with id ${id}`);
     }
 
+    let transportType: RelayerTypes.TransportType = "relay";
+    let appLink;
+    if (this.isLinkModeEnabled(pendingRequest.requester.metadata)) {
+      transportType = "link_mode";
+      appLink = pendingRequest.requester.metadata.redirect?.universal;
+    }
+
```

Comment:
can we move these inside the `sendError, sendRequest` instead of setting this on every usage e.g. if link mode is enabled etc etc to be used unless disabled by flag, for example on `disconnect` requests

Reply:
Okey, you mean to add a disable flag param to sendError, sendRequest?

`sendRequest({...args, disableLinkMode: true})`

Reply:
or define an array of methods that don't support linkMode, and add a condition inside `sendError`, `sendRequest` and `sendResult`